rust-runtime/aws-smithy-http-server/src/plugin/mod.rs

```diff
@@ -3,6 +3,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+//! The plugin system allows for parameterized modification of [`Operation`]s during the build procedure.
```

Comment:
This feels a bit abstract 🤔

Reply:
Yeah, any thoughts?

Perhaps a doc test with some annotations will make it more concrete?

Reply:
A doc test would certainly help.

Perhaps

```
The plugin system allows you to build middlewares that have access to the details of the [`Operation`] that was invoked (e.g. its name).
```

Reply:
Hows this look?

https://github.com/awslabs/smithy-rs/pull/2020/commits/d717444a0f0fd5e9c82edca190a217c2168e6d2b

I'm wondering whether it'd be nice to demonstrate the extension trait? Should I pull in `InstrumentExt` in an example?

Reply:
That looks good!
I would avoid duplicating too much content between the module docs and the `PluginPipeline` docs.